### PR TITLE
Add PodFetch

### PR DESCRIPTION
**Pull request by Rabenherz112**
Thanks for taking the time to suggest an addition to awesome-selfhosted! 

To ensure your Pull Request is dealt with swiftly, please check the following (check the boxes `[x]`):
- [X] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [X] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted-data/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pulls), including closed ones.
- [X] Any software you are adding is not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [X] The file you are adding is formatted as described in [addition.md](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/ISSUE_TEMPLATE/addition.md).
- [X] `Demo` links should only be used for interactive demos, i.e. not video demonstrations. 
- [X] Comments and unused optional fields have been removed.
- [X] The file you are adding uses [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) file naming, for example `my-awesome-software.yml`.
- [X] Values for `platform` are the main **server-side** requirements for the software. Don't include frameworks or specific dialects.
- [X] Any software project you are adding to the list is actively maintained.
- [X] Any software project you are adding was first released more than 4 months ago.
- [X] Any software project you are adding has working installation instructions.
- [X] You understand that your Pull Request will be merged ~1 week after approval, to allow for further comments if needed.

### CODE REVIEW

**Review by kokomo123 — APPROVED** [review]
Mature license, looks accurate, LGTM. No issues here, pending other thoughts from collaborators. 👍🏻

File: software/podfetch.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+website_url: "https://samtv12345.github.io/PodFetch"
+source_code_url: "https://github.com/SamTV12345/PodFetch"
+description: "A sleek and efficient podcast downloader."
+licenses:
+  - Apache-2.0
+platforms:
+  - Docker
+  - Rust
+tags:
+  - Automation
```
Comment (nodiscc) [inline software/podfetch.yml:11]: Other podcast-related software is under https://awesome-selfhosted.net/tags/media-streaming---audio-streaming.html
Shouldn't we use this tag/category here as well?

Comment (kokomo123) [reply software/podfetch.yml:11]: yeah, that seems about right. Forgot we had that tag.

Comment (kokomo123) [inline software/podfetch.yml:11]:
```suggestion
  - Automation
  - Media Streaming - Audio Streaming
```
Here ya are raben

Comment (Rabenherz112) [reply software/podfetch.yml:11]: Thank you :)

**Review by kokomo123 — APPROVED** [review]

**Pull request merged**